api/storage_service.cc

```diff
@@ -77,6 +77,18 @@ sstring validate_keyspace(http_context& ctx, const parameters& param, sstring pa
     throw bad_param_exception(fmt::format("Keyspace '{}' does not exist", ks_name));
 }
 
+std::vector<sstring> validate_tables(const sstring& ks_name, http_context& ctx, const parameters& param, sstring param_name) {
```

Comment:
> So again we go back to the wrong exception...

I don't think it's wrong to return internal_server_error code if the tables exist when the api function is called and later disappears.  It's certainly can't be considered a bad request.  If it's retried on the client, the retry invocation will get the bad_request error since now the tables are gone.

> So isn't it better to just add the validation inside the loop that works on each individual table, and not as an advanced step going over all the tables?

In the case that the user provided no tables option we get the list of tables from the database, and they can essentially go away too if this is done with a DROP TABLE in parallel.

I think it's going to be super weird for a user to run an operation on the whole keyspace and get a bad_request error for an internal error like this.

The right thing to do is to lock the keyspace in such a way that will prevent tables from being dropped in the middle of the operation - OR provide a way to abort the operation internally if a table is dropped during an api operation, log that, and return an appropriate error to the user.

I think that for short-running operations like enable/disable auto_compaction, locking is fine.
For long-running operations, like running cleanup compaction, we need first to convert the api to asynchronous (generalization of https://github.com/scylladb/scylla/issues/8392) and then the operation can be managed more carefully and be aborted (table removal does stop ongoing compactions, including cleanup, I think this need not generate an error if there's a background cleanup operation on it)

Reply:
Done in #9755